internal/backend/usagereporter/usagedata.go

```diff
@@ -0,0 +1,145 @@
+package usagereporter
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/google/uuid"
+	"go.autokitteh.dev/autokitteh/internal/version"
+	"go.autokitteh.dev/autokitteh/internal/xdg"
+	"go.uber.org/zap"
+)
+
+type reportRequest struct {
+	InstallationID string `json:"installation_id"`
+	Version        string `json:"version"`
+	Commit         string `json:"commit"`
+	OS             string `json:"os"`
+	Arch           string `json:"arch"`
+}
+
+type UsageReporter interface {
+	Start()
+	Stop()
+}
+
+type usageReporter struct {
+	installationID uuid.UUID
+	updateInterval time.Duration
+	shutdownChan   chan struct{}
+	endpoint       string
+	logger         *zap.Logger
+}
+
+func generateInstallIDFile(path string) error {
+	id := uuid.New()
```

Comment:
Question: did you intentionally choose UUIDv4? Any opinion about which UUID is better for this purpose - v4 (random) or v7 (timestamp + random)?

Reply:
doesn't really matter, can also be shortid or something, since each machine can have different clock, we won't really get anything from the timestamp
it just needs to be random

Reply:
sgtm